geopandas/base.py

```diff
@@ -4588,6 +4588,94 @@ def sample_points(self, size, method="uniform", seed=None, rng=None, **kwargs):
+
+        Polygonizes the GeoSeries that contain linework which represents the
+        edges of a planar graph. Any geometry type may be provided as input; only the
+        constituent lines and rings will be used to create the output polygons.
+
+        Lines or rings that when combined do not completely close a polygon will be
+        ignored. Duplicate segments are ignored.
+
+        Unless you know that the input GeoSeries represents clean topological of a
+        planar graph (e.g. there is a node on both lines where they intersect), it is
+        recommended to use ``node=True`` which performs noding prior polygonization.
+        Using ``node=False`` will provide performance benefits but may result in
+        incorrect polygons if the input is not of the proper topology.
+
+        When ``full=True``, the return value is a 4-tuple containing output polygons,
+        along with lines which could not be converted to polygons. The return value
+        consists of 4 elements:
+
+            - GeoSeries of the valid polygons (same as with ``full=False``)
+            - GeoSeries of cut edges: edges connected on both ends but not part of
+              polygonal output
+            - GeoSeries of dangles: edges connected on one end but not part of polygonal
+              output
+            - GeoSeries of invalid rings: polygons that are formed but are not valid
+              (bowties, etc)
+
+        Parameters
+        ----------
+        node : bool, optional
+            perform noding prior polygonization, by default True
+        full : bool, optional
+            return the full output composed of a tuple of GeoSeries, by default False
+
+        Returns
+        -------
+        GeoSeries | tuple(GeoSeries, GeoSeries, GeoSeries, GeoSeries)
+            GeoSeries with the polygons or a tuple of four GeoSeries as
+            ``(polygons, cuts, dangles, invalid)``
+
+        Examples
+        --------
+        >>> from shapely.geometry import LineString
+        >>> s = geopandas.GeoSeries([
+        ...     LineString([(0, 0), (1, 1)]),
+        ...     LineString([(0, 0), (0, 1), (1, 1), (1, 0), (0, 0)]),
+        ...     LineString([(0.5, 0.2), (0.5, 0.8)]),
+        ... ])
+        >>> s.polygonize()
+        0        POLYGON ((0 0, 0.5 0.5, 1 1, 1 0, 0 0))
+        1    POLYGON ((0.5 0.5, 0 0, 0 1, 1 1, 0.5 0.5))
+        Name: polygons, dtype: geometry
+
+        >>> polygons, cuts, dangles, invalid = s.polygonize(full=True)
+
+        """
+        from .geoseries import GeoSeries
+        from .array import from_shapely
+
+        if node:
+            geometry_input = from_shapely([self.geometry.unary_union])
+        else:
+            geometry_input = self.geometry.values
+
+        if full:
+            polygons, cuts, dangles, invalid = shapely.polygonize_full(geometry_input)
+
+            cuts = GeoSeries(cuts, crs=self.crs, name="cut edges").explode(
```

Comment:
```suggestion
            cuts = GeoSeries(cuts, crs=self.crs, name="cut_edges").explode(
```

Just an idea, but in general names without spaces is better practice?